controllers/gateway_kuadrant_controller.go

```diff
@@ -119,18 +102,67 @@ func (r *GatewayKuadrantReconciler) reconcileKuadrantNamespaceAnnotation(ctx con
 			keys[idx] = client.ObjectKeyFromObject(&kuadrantList.Items[idx]).String()
 		}
 		logger.Info("Multiple kuadrant instances found", "num", len(kuadrantList.Items), "keys", strings.Join(keys[:], ","))
-		return false, nil
+		return nil
 	}
 
-	kuadrant.AnnotateObject(gw, kuadrantList.Items[0].Namespace)
+	if len(kuadrantList.Items) == 0 {
+		logger.Info("Kuadrant instance not found in the cluster")
+		return r.removeKuadrantNamespaceAnnotation(ctx, gw)
+	}
+
+	val, ok := gw.GetAnnotations()[kuadrant.KuadrantNamespaceAnnotation]
+	if !ok || val != kuadrantList.Items[0].Namespace {
+		// either the does not exist or is different, hence update
```

Comment:
```suggestion
		// Either the annotation does not exist or the namespaces differs from the available Kuadrant CR, hence we update
```

Reply:
I changed a little bit from your suggestion. Let me know what you think

Reply:
Much better! <3